### PR TITLE
compute: enrich MatchContext with simple repo information

### PR DESCRIPTION
**Pull request by coury-clark**
Closes https://github.com/sourcegraph/sourcegraph/issues/34228

Enrich the compute stream API MatchContext result with simple repository information (repository ID and repository name)

## Test plan
Here's an example event from the stream running locally
```
event: results
data: [{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestFindMatchingSeries/find_a_matching_series_when_one_exists.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}}}}}],"path":"enterprise/internal/insights/background/queryrunner/testdata/TestGetSeries/series_definition_does_exist.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestFindMatchingSeries/match_capture_group_series.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":5,"column":7},"end":{"offset":-1,"line":5,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":5,"column":7},"end":{"offset":-1,"line":5,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateSeries/test_create_and_get_capture_groups_series.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateGetView_WithGrants/no_users_or_orgs_provided_should_only_return_global.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateGetView_WithGrants/org_5_can_see_the_view.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateGetView_WithGrants/user_1_can_see_this_view.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"}]
```

### CODE REVIEW

File: enterprise/internal/compute/match_context_result.go [hunk 1]
```diff
@@ -34,8 +34,10 @@ type Match struct {
 }
 
 type MatchContext struct {
-	Matches []Match `json:"matches"`
-	Path    string  `json:"path"`
+	Matches    []Match `json:"matches"`
+	Path       string  `json:"path"`
+	RepoID     int32   `json:"repositoryID"`
```
Comment (rvantonder) [inline enterprise/internal/compute/match_context_result.go:39]: Nit: can we take the opportunity to call this `RepositoryID`? I know the it's coming from `fm.Repo.ID` but let's expand this. We do so elsewhere with external-bound field names: https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+repositoryid+lang:go&patternType=literal

**Review by rvantonder — APPROVED** [review]

File: enterprise/internal/compute/match_context_result.go (hunk 1 above)
Comment (coury-clark) [reply enterprise/internal/compute/match_context_result.go]: Sure! On latest:

```
event: results
data: [{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestFindMatchingSeries/find_a_matching_series_when_one_exists.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}}}}}],"path":"enterprise/internal/insights/background/queryrunner/testdata/TestGetSeries/series_definition_does_exist.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":6,"column":7},"end":{"offset":-1,"line":6,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestFindMatchingSeries/match_capture_group_series.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":5,"column":7},"end":{"offset":-1,"line":5,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":5,"column":7},"end":{"offset":-1,"line":5,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateSeries/test_create_and_get_capture_groups_series.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateGetView_WithGrants/no_users_or_orgs_provided_should_only_return_global.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateGetView_WithGrants/org_5_can_see_the_view.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"},{"matches":[{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}},"environment":{"1":{"value":"Historical","range":{"start":{"offset":-1,"line":8,"column":7},"end":{"offset":-1,"line":8,"column":17}}}}}],"path":"enterprise/internal/insights/store/testdata/TestCreateGetView_WithGrants/user_1_can_see_this_view.golden","repositoryID":11,"repository":"github.com/sourcegraph/sourcegraph"}]

```

Comment (rvantonder) [reply enterprise/internal/compute/match_context_result.go]: ty

**Pull request merged**